File: web/tests/reference/sorting/largest-number.py
from functools import cmp_to_key


def largest_number(nums):
    strs = [str(n) for n in nums]

    def compare(a, b):
        if a + b > b + a:
            return -1
        if a + b < b + a:
            return 1
        return 0

    strs.sort(key=cmp_to_key(compare))
    result = "".join(strs)
    return "0" if result[0] == "0" else result

File: web/tests/reference/sorting/largest-number.js
function largestNumber(nums) {
  const strs = nums.map((n) => String(n));

  strs.sort((a, b) => {
    if (a + b > b + a) return -1;
    if (a + b < b + a) return 1;
    return 0;
  });

  const result = strs.join("");
  return result[0] === "0" ? "0" : result;
}
